### PR TITLE
feat: adding entry about mdx

### LINKED ISSUE #2872: [Term Entry] Subject: MDX
### Reason/inspiration (optional)

I've been working with MDX lately. It's an extension to Markdown that lets people include JSX in Markdown documents, and I would love to add this as a concept entry to the Markdown section. 

### Entry Type (select all that apply)

New Concept Entry for Existing Topic

### Code of Conduct

- [X] By submitting this issue, I agree to follow Codecademy Doc's [Code of Conduct](https://github.com/Codecademy/docs/blob/main/.github/CODE_OF_CONDUCT.md).

### For Maintainers

- [x] Labels added
- [x] Issue is assigned

### PR DESCRIPTION
**Pull request by CBID2**
### Description
This PR adds a new file called `mdx.md`, which explains the concept of MDX.
Closes #2872 

### Type of Change

- Adding a new entry

### Checklist

- [X] All writings are my own.
- [X] My entry follows the Codecademy Docs style guide.
- [ ] My changes generate no new warnings.
- [X] I have performed a self-review of my own writing and code.
- [X] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [X] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [X] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

### CODE REVIEW

**Comment by CBID2** [comment]
Hi @KTom101! :) I'm ready for a review! :)

File: content/markdown/concepts/mdx/mdx.md [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+  - 'paths/front-end-engineer-career-path'
+---
+
+**MDX** is an extension to Markdown that allows developers to include JSX and import components like interactive charts and alerts to Markdown documents that contain long content more engaging.
+
+## Syntax
+
+There is not specific set standard for adding MDX, so for the sake of this entry, the information below shows pseudo MDX code:
+
+```mdx
```
Comment (KTom101) [inline content/markdown/concepts/mdx/mdx.md:21]:
```suggestion
```pseudo
```

File: content/markdown/concepts/mdx/mdx.md [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+const MyMDX = () => {
+return (
+<>
+<h1>This is a heading</h1>
+<MyComponent />
+</>
+);
+};
+
+export default MyMDX;
```
Comment (KTom101) [inline content/markdown/concepts/mdx/mdx.md:38]: Don't forget to close the block (```)

File: content/markdown/concepts/mdx/mdx.md [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+## Example
+
+In this example, the variable, `cookies`, is used throughout the MDX file to provide explanations to children on why they are delicious.
+
+md
```
Comment (KTom101) [inline content/markdown/concepts/mdx/mdx.md:44]:
```suggestion
```md
```

File: content/markdown/concepts/mdx/mdx.md [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+  - 'Web Development'
+Tags:
+  - 'Formatting'
+  - 'Documentation'
+CatalogContent:
+  - 'learn-html'
+  - 'paths/front-end-engineer-career-path'
+---
+
+**MDX** is an extension to Markdown that allows developers to include JSX and import components like interactive charts and alerts to Markdown documents that contain long content more engaging.
```
Comment (KTom101) [inline content/markdown/concepts/mdx/mdx.md:15]:
```suggestion
**MDX** is an extension to Markdown that allows developers to include JSX and import components like interactive charts and alerts to Markdown documents that contain lengthy content more engaging.
```

**Review by KTom101 — CHANGES_REQUESTED** [review]
@CBID2 I added a few changes

**Comment by CBID2** [comment]
> @CBID2 I added a few changes 

Done @KTom101

File: content/markdown/concepts/mdx/mdx.md [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+return (
+<>
+<h1>This is a heading</h1>
+<MyComponent />
+</>
+);
+};
+
+export default MyMDX;
+```
```
Comment (KTom101) [inline content/markdown/concepts/mdx/mdx.md:39]:
```suggestion
```

```

**Review by KTom101 — APPROVED** [review]
@CBID2 Just one more change but this is ready for a 2nd round of review.

File: content/markdown/concepts/mdx/mdx.md [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+  - 'Web Development'
+Tags:
+  - 'Formatting'
+  - 'Documentation'
+CatalogContent:
+  - 'learn-html'
+  - 'paths/front-end-engineer-career-path'
+---
+
+**MDX** is an extension to Markdown that allows developers to include JSX and import components like interactive charts and alerts to Markdown documents that contain lengthy content more engaging.
```
Comment (SSwiniarski) [inline content/markdown/concepts/mdx/mdx.md:15]:
```suggestion
**MDX** is an extension to Markdown that allows developers to include JSX and import components like interactive charts and alerts to make Markdown documents that contain lengthy content more engaging.
```

**Review by SSwiniarski — CHANGES_REQUESTED** [review]
@CBID2 I did a review and had one comment.

**Comment by CBID2** [comment]
> @CBID2 I did a review and had one comment.

Done @SSwiniarski

**Review by SSwiniarski — APPROVED** [review]
@CBID2 Looks good.

**Pull request merged**